modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/size_items.dm

```diff
@@ -0,0 +1,129 @@
+/// What areas are we allowed to use size items in?
+#define SIZE_WHITELISTED_AREAS list(\
+		/area/centcom/interlink/dorm_rooms,\
+		/area/centcom/holding/cafedorms,\
+		/area/misc/hilbertshotel,\
+)
+
+/obj/item/clothing/neck/size_collar
+	name = "size collar"
+	desc = "A shiny black collar embeded with technology that allows the user to change their own size."
+	icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/obj/lewd_clothing/lewd_neck.dmi'
+	worn_icon = 'modular_skyrat/modules/modular_items/lewd_items/icons/mob/lewd_clothing/lewd_neck.dmi'
+	icon_state = "collar_black"
+	/// Have we given the user the warning message yet?
+	var/warning_given = FALSE
+	/// The `temporary_size` component we have attached to the wearer.
+	var/datum/component/temporary_size/size_component
+	/// What size do we want to set the wearer to when they wear the collar?
+	var/target_size = 1
+
+/obj/item/clothing/neck/size_collar/attack_self(mob/user, modifiers)
+	. = ..()
+	if(!warning_given)
+		if(tgui_alert(user, "This item is strictly intended as an ERP item for use in dorm rooms. Failure to respect this will result in administrative action being taken. Do you wish to continue using this item?", "A word of warning.", list("Yes", "No")) != "Yes")
+			return FALSE
+
+		warning_given = TRUE
+
+	var/chosen_size = tgui_input_number(user, "What size percentage do you wish to set the collar to?", name, 100, CONFIG_GET(number/size_collar_maximum), CONFIG_GET(number/size_collar_minimum))
+	if(!chosen_size)
+		balloon_alert(user, "invalid size!")
+		return FALSE
+
+	log_message("[src] had its target size changed to [chosen_size]% by [usr]", LOG_ATTACK)
+	balloon_alert(user, "set to [chosen_size]%")
+	target_size = (chosen_size * 0.01)
```

Comment:
I'm sure we got a define for converting something away from percents, don't we? This multiplication could otherwise be a little confusing.

Reply:
marking this as resolved, I'll try and bring this upstream at a later date.